src/library/scala/collection/mutable/Map.scala

```diff
@@ -197,8 +199,9 @@ trait MapOps[K, V, +CC[X, Y] <: MapOps[X, Y, CC, _], +C <: MapOps[K, V, CC, C]]
     * @return   the map itself.
     */
   def mapValuesInPlace(f: (K, V) => V): this.type = {
-    if (nonEmpty) this match {
+    if (!isEmpty) this match {
```

Comment:
Why `!isEmpty` instead of nonEmpty?

Reply:
basically, despite the fact that the latter really ought to be always trivially optimisable, `!isEmpty` sometimes has better performance then `nonEmpty`. I don't recall when/where this discovery happened or was mentioned; the best I can find in reference to it is #7898